db/sql/0_msar.sql

```diff
@@ -2237,6 +2295,122 @@ END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
+-- Comment on column -------------------------------------------------------------------------------
+
+
+CREATE OR REPLACE FUNCTION
+__msar.comment_on_column(
+  tab_name text,
+  col_name text,
+  comment_ text
+) RETURNS text AS $$/*
+Change the description of a column, returning command executed. If comment_ is NULL, column's
+comment is removed.
+
+Args:
+  tab_name: The name of the table containg the column whose comment we will change.
+  col_name: The name of the column whose comment we'll change
+  comment_: The new comment. Any quotes or special characters must be escaped.
+*/
+DECLARE
+  comment_or_null text := COALESCE(comment_, 'NULL');
+BEGIN
+RETURN __msar.exec_ddl(
+  'COMMENT ON COLUMN %s.%s IS %s',
+  tab_name,
+  col_name,
+  comment_or_null
+);
+END;
+$$ LANGUAGE plpgsql;
+
+
+CREATE OR REPLACE FUNCTION
+msar.comment_on_column(
+  sch_name text,
+  tab_name text,
+  col_name text,
+  comment_ text
+) RETURNS text AS $$/*
+Change the description of a column, returning command executed.
+
+Args:
+  sch_name: The schema of the table whose column's comment we will change.
+  tab_name: The name of the table whose column's comment we will change.
+  col_name: The name of the column whose comment we will change.
+  comment_: The new comment.
+*/
+SELECT __msar.comment_on_column(
+  msar.get_fully_qualified_object_name(sch_name, tab_name),
+  quote_ident(col_name),
+  quote_literal(comment_)
+);
+$$ LANGUAGE SQL;
+
+
+CREATE OR REPLACE FUNCTION
+__msar.comment_on_column(
+  tab_id oid,
+  col_name text,
+  comment_ text
+) RETURNS text AS $$/*
+Change the description of a column, returning command executed.
+
+Args:
+  tab_id: The OID of the table containg the column whose comment we will change.
+  col_name: The name of the column whose comment we'll change
+  comment_: The new comment. Any quotes or special characters must be escaped.
+*/
+SELECT __msar.comment_on_column(
+  __msar.get_relation_name(tab_id),
+  col_name,
+  comment_
+);
+$$ LANGUAGE SQL;
+
+
+CREATE OR REPLACE FUNCTION
+__msar.comment_on_column(
+  tab_id oid,
+  col_id integer,
+  comment_ text
+) RETURNS text AS $$/*
+Change the description of a column, returning command executed.
+
+Args:
+  tab_id: The OID of the table containg the column whose comment we will change.
+  col_id: The ATTNUM of the column whose comment we will change.
+  comment_: The new comment.
+*/
+SELECT __msar.comment_on_column(
+  __msar.get_relation_name(tab_id),
+  msar.get_column_name(tab_id, col_id),
+  comment_
+);
+$$ LANGUAGE SQL;
```

Comment:
This function could be removed, in combination with the other comment.

Reply:
I left this one around. I made the analogue public function be based on this one instead.

Reply:
Relevant commits https://github.com/centerofci/mathesar/pull/3186/commits/4bdc9b8c43a42e83dc5868686a29ff60cbfbd5c1 and https://github.com/centerofci/mathesar/pull/3186/commits/625b868f51376b3908f5a0df7438ccc4d2119008

Reply:
Relevant commits https://github.com/centerofci/mathesar/pull/3186/commits/4bdc9b8c43a42e83dc5868686a29ff60cbfbd5c1 and https://github.com/centerofci/mathesar/pull/3186/commits/625b868f51376b3908f5a0df7438ccc4d2119008